app/models/activity.rb

```diff
@@ -23,6 +23,7 @@ class Activity < ApplicationRecord
   belongs_to :organisation
   belongs_to :extending_organisation, foreign_key: "extending_organisation_id", class_name: "Organisation", optional: true
   has_many :implementing_organisations
+  belongs_to :reporting_organisation, foreign_key: "reporting_organisation_id", class_name: "Organisation"
```

Comment:
I think mine are even worse tbh, I think I added this because it was a cut & paste from elsewhere.

Reply:
@tahb I think this is the issue you refer to in https://github.com/UKGovernmentBEIS/beis-report-official-development-assistance/pull/297#issuecomment-610472101 - before we introduce another db constraint shall we talk it over - now feels like the time to sort this if we switch to a solution in the app code?

Reply:
I definitely agree. Now that we're about to add our second constraint is a great time to review what our approach is. I think the right thing to do is to add constraints in rather than take them away. I will review and start a conversation with you both as soon as I can.

Reply:
I've opened a pull request and started a conversation here https://github.com/UKGovernmentBEIS/beis-report-official-development-assistance/pull/321. I propose that this work is blocked pending the outcome of that convo?

Reply:
Yeah it will be. This PR is going to haunt me forever!!

Reply:
Mec isn't working today and it'd be great to unblock this. Suggesting we merge this as 2 of us feel adding constraints is the better way to go. The cost to reverting that decision to go the other way is low since removing constraints is much easier than adding them!